jupyterhub/handlers/login.py

```diff
@@ -90,13 +90,20 @@ class LoginHandler(BaseHandler):
     """Render the login page."""
 
     def _render(self, login_error=None, username=None):
+        # login url with any parameters specified by the user/application
+        login_url = self.append_query_parameters(
+            self.settings['login_url'], exclude=['next']
+        )
+        # the next url for after the login
+        next_url = url_escape(self.get_argument('next', default=''))
```

Comment:
I don't think it's quite right that the URL parameters from the login page are preserved on the destination. Instead, they should be *encoded* in the `next` URL. Adding them here might be doubly-encoding params.

Reply:
Ah! I think that makes sense! Let me try that.

Reply:
Had a try at adding the tests and adding the parameters escaped/encoded to the `next` parameter @minrk . Let me know if that looks better :+1: